### ISSUE TITLE
Capture image was flipped horizontal and had green screen

### ISSUE THREAD

**Issue by AnhPC03** [Hugand/camera_tutorial#6]
Thanks for your public code.
I ran your code and had some issues that the image captured by Camera was flipped and had screen
![Screenshot_2021-01-07-17-20-28](https://user-images.githubusercontent.com/35995951/103881296-d5998e80-510c-11eb-9ea9-e3b62b8e21b7.png)
![Screenshot_2021-01-07-17-20-19](https://user-images.githubusercontent.com/35995951/103881300-d6cabb80-510c-11eb-89b4-81ad14f9291e.png)

**Comment by ugu11** [comment]
Interesting. I'll need to look into it then. It might be the C code that's causing it

**Comment by AnhPC03** [comment]
@Hugand thanks for your fast reply.
I had checked and realized that if i used android 7 or higher. It worked well excepted having green bar at the bottom, and that bar pushed the image. It seemed that the bar wasn't a part of image.
If i used android 5, then the result was this issue i opened (green screen and flipped)

**Comment by assanbayev** [comment]
@Hugand please can take a look in that issue. I have the same issue

**Comment by ugu11** [comment]
@assanbayev Yes of course. Can you provide more details on the device you're using and the android version?

**Comment by assanbayev** [comment]
@Hugand Hi. Thank you for responce. On xiaomi divices image has green bottom bar. on some huawei with android 8, galaxy s20 with android 11, samsung a51 with android 10 works well. Bun on xiaomi note 8,note 9, and on some redmi with android 9,10 it has the issue. on working devices u or v array length is equal to ((y array length)/2 -1) and on not working devices there's lack of u and v array bytes. For example in working image y = 307200, u = 153599, v = 153599 and in not working image y = 168592, u = 184271, v = 184271 
![WhatsApp Image 2021-03-25 at 7 53 45 AM](https://user-images.githubusercontent.com/38101765/112412835-5d288e00-8d49-11eb-910f-d7d59c88c3c7.jpeg)

**Comment by assanbayev** [comment]
@Hugand Why there's additional bytes on xiaomi devices? I  mean 480*720=345600 and its not what we have in y array 368592. I guess thats the problem
